### PR TITLE
Add search plugin for PTP

### PR DESCRIPTION
**Pull request by Chemondelay**
Configured like so:
```
discover:
  ...
  from:
    - ptp:
        username: *****
        password: *****
        passkey: *****   # Found after `torrent_pass=` in any torrent download link on the site
```

The plugin populates the usual entry fields, as well as three unique boolean fields that can be filtered against:
 * `ptp_checked`, indicating whether the torrent has been checked by staff;
 * `ptp_gp`, indicating a 'Golden Popcorn' (i.e. verified high-quality) release; and
 * `ptp_scene`, indicating a Scene release.

### CODE REVIEW

File: flexget/plugins/search_ptp.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+from flexget import plugin
+from flexget.entry import Entry
+from flexget.event import event
+from flexget.utils import requests, json
+from flexget.utils.search import torrent_availability
+
+log = logging.getLogger('ptp')
+
+session = requests.Session()
```
Comment (stevezau) [inline flexget/plugins/search_ptp.py]: You should be able to use 

task.session and task.requests

File: flexget/plugins/search_ptp.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            'username': {'type': 'string'},
+            'password': {'type': 'string'},
+            'passkey': {'type': 'string'}
+        },
+        'required': ['username', 'password', 'passkey'],
+        'additionalProperties': False
+    }
+
+    def search(self, task, entry, config):
+        session.post(
```
Comment (stevezau) [inline flexget/plugins/search_ptp.py]: Perhaps it should check for failed credentials?

File: flexget/plugins/search_ptp.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+        try:
+            r = session.post('%s/torrents.php?json=1&searchstr=%s' % (base_url, search))
+        except RequestException as e:
+            log.error('Error searching PTP: %s' % e)
+            return
+        try:
+            content = json.loads(r.content)
+        except ValueError as e:
+            log.debug('No results from PTP.')
```
Comment (stevezau) [inline flexget/plugins/search_ptp.py]: json.loads(r.content) should work even if it's empty json.. I guess this should say invalid response from server or something similar?

File: flexget/plugins/search_ptp.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            return
+
+        results = set()
+        if 'Torrents' in content:
+            for item in content['Torrents']:
```
Comment (stevezau) [inline flexget/plugins/search_ptp.py]: could just do 

for item in content.get('Torrents', []):

File: flexget/plugins/search_ptp.py (hunk 3 above)
Comment (nikdoof) [inline flexget/plugins/search_ptp.py]: This has probably been done to catch error 500s and cloud flare issues, but this isn't the right way to do it. We should be checking ```r.status_code``` and making use of ```r.json```, let requests do the work for us.

**Comment by liiight** [comment]
@Chemondelay is this still relevant?

**Comment by liiight** [comment]
Reopen if relevant

**Pull request closed without merging**